### ISSUE TITLE
What do you mean "resumable"? jQuery File Upload doesn't seem to be resumable.

### ISSUE THREAD

**Issue by cawecoy** [pionl/laravel-chunk-upload#109]
Hi. I really appreciate this work, thank you first of all.

I was using this solution with jQuery File Upload and followed all the instructions in this repository.

But when I reload the page in the middle of the upload and try to upload again, the chunk file is deleted and the upload restart from 0% again.

According to my understanding, it should reuse what was previously uploaded before I reload the page.

Just a newbie question, maybe I don't get what does "resumable" means.

**Comment by nerg4l** [comment]
> ## Features
> 
> [...]
> - **Resumable uploads:**  
>   Aborted uploads can be resumed with browsers supporting the Blob API.
> [...]

Source: https://github.com/blueimp/jQuery-File-Upload#features

To see what you need to do on JavaScript side check out this link: https://github.com/blueimp/jQuery-File-Upload/wiki/Chunked-file-uploads#resuming-file-uploads

**Comment by cawecoy** [comment]
@nerg4l thank you!

The link from the jQuery File Upload says:

> The above code overrides the add callback and sends a JSON request with the current file name to the server. If a file with the given name exists, the server responds with the file information including the file size, which is set as uploadedBytes option.

I can't find any code in Laravel Chunk Upload repository handling this case. Or am I missing something?

As I can see, `FileReceiver` constructor doesn't even create a handler in case there is no uploaded file in the resquest https://github.com/pionl/laravel-chunk-upload/blob/f5e24dd587b20cb4f303a79069b94eece94aae9d/src/Receiver/FileReceiver.php#L66

I guess it is because the handler are not made for this case, they expect to have a file uploaded, not only a filename, i.e. https://github.com/pionl/laravel-chunk-upload/blob/f5e24dd587b20cb4f303a79069b94eece94aae9d/src/Handler/AbstractHandler.php#L95

So I developed a workaround: since chunk file is only one file by the jQuery File Upload logic, I just rebuild the chunk file name as the `createChunkFileName` method from `AbstractHandlerdoes` class does and check if this chunk file exists.

I also did changes in the jQuery Upload File's javascript example to send the file size as it is part of the chunk file name: `fileSize: data.files[0].size`

https://gist.github.com/cawecoy/b8d02dfe99c29b69e3ecdb8ed2b868d6
